src/System.Text.Primitives/System/Text/TextEncoding.cs

```diff
@@ -121,6 +123,156 @@ public override bool TryEncodeFromUtf16(ReadOnlySpan<char> utf16, Span<byte> buf
+            int bytesWrittenForCodePoint = 0;
+            bytesWritten = 0;
+
+            for (int i = 0; i < decoded.Length; i++)
+            {
+                UnicodeCodePoint decodedCodePoint = decoded[i];
+                if (!TryDecodeToUnicode(encoded.Slice(bytesWritten), out decodedCodePoint, out bytesWrittenForCodePoint))
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+                decoded[i] = decodedCodePoint;
+                bytesWritten += bytesWrittenForCodePoint;
+                if (bytesWritten > avaliableBytes)
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public override bool TryDecodeToUnicode(Span<byte> encoded, out UnicodeCodePoint decoded, out int bytesWritten)
+        {
+            bytesWritten = 0;
+
+            if (encoded.IsEmpty)
+            {
+                decoded = new UnicodeCodePoint();
+                return false;
+            }
+            
+            byte firstByte = encoded[0];
+
+            bytesWritten = CountConsecutiveStartingOnes(firstByte);
+            if (bytesWritten == 0) bytesWritten++;
+
+            uint answer = firstByte;
+
+            if (bytesWritten == 1)
+            {
+                // Do nothing
+            }
+            else if (bytesWritten == 2)
+            {
+                byte byte0 = (byte)(firstByte & 0x1F);
+                byte byte1 = (byte)(encoded[1] & 0x3F);
+                answer = (uint)(((byte0 >> 2) << 8) | ((byte0 << 6) | byte1));
+            }
+            else if (bytesWritten == 3)
+            {
+                byte byte0 = (byte)(firstByte & 0x0F);
+                byte byte1 = (byte)(encoded[1] & 0x3F);
+                byte byte2 = (byte)(encoded[2] & 0x3F);
+                answer = (uint)((((byte0 << 4) | (byte1 >> 2)) << 8) | ((byte1 << 6) | byte2));
+            }
+            else if (bytesWritten == 4)
+            {
+                byte byte0 = (byte)(firstByte & 0x07);
+                byte byte1 = (byte)(encoded[1] & 0x3F);
+                byte byte2 = (byte)(encoded[2] & 0x3F);
+                byte byte3 = (byte)(encoded[3] & 0x3F);
+                answer = (uint)((((byte0 << 2) | (byte1 >> 4)) << 16) | (((byte1 << 4) | (byte2 >> 2)) << 8) | ((byte2 << 6) | byte3));
+            }
+
+            decoded = new UnicodeCodePoint(answer);
+            return true;
+        }
+
+        private int CountConsecutiveStartingOnes(byte input)
+        {
+            int count = 0;
+            while ((input & (1 << 7)) != 0)
+            {
+                input = (byte)(input << 1);
+                count++;
+            }
+            return count;
+        }
+
         public override bool TryEncodeString(string value, Span<byte> buffer, out int bytesWritten)
         {
             var avaliableBytes = buffer.Length;
```

Comment:
Don't we already have that here:
https://github.com/dotnet/corefxlab/blob/master/src/System.Text.Primitives/System/Text/TextEncoding.cs#L45

So, should we just remove the override methods?